index.js

```diff
@@ -10,6 +10,7 @@ function generate(argIndex) {
 		var promise = thunk(value);
 		thunk.then = promise.then.bind(promise);
 		thunk.catch = promise.catch.bind(promise);
+		thunk._actualPromise = promise;
```

Comment:
This exposes one potential flaw in our strategy of attaching the promise methods to the thunk. The `unhandledRejection` reports the underlying promise, not the `thunk` that we turned into a Promise.

I almost think the `unhandledRejection` spec should be expanded to handle this sort of situation:

```js
promise[Symbol.for('unhandledRejectionSurrogate')] = thunk;
```

Which would cause the surrogate (`thunk` in this case) to be reported instead.